src/mui/input/DateInput.js

```diff
@@ -18,7 +18,7 @@ export const datify = input => {
 
 class DateInput extends Component {
     onChange = (_, date) => {
-        this.props.input.onChange(date);
+        this.props.input.onChange(date.toISOString());
```

Comment:
If we return a date here, it will be serialized for the url, but stay a date in the state. And if we go in the menu, and then comeback, the restClient will receive the date from the state.
On the contrary, if we come by entering the url, the restClient will receive the date as a string from the url.
This ensure that the date is stored as a string both for the url and the state.